docs/src/main/asciidoc/kafka.adoc

```diff
@@ -629,6 +633,76 @@ public class FruitResource {
 }
 ----
 
+== Avro serialization
+
+This is described in a dedicated guide: link:kafka-schema-registry-avro.adoc[Using Apache Kafka with Schema Registry and Avro].
+
+[[serialization-autodetection]]
+== Serializer/deserializer autodetection
+
+When using SmallRye Reactive Messaging with Kafka, Quarkus can often automatically detect the correct serializer and deserializer class.
+This autodetection is based on declarations of `@Incoming` and `@Outgoing` methods, as well as injected ``@Channel``s.
+
+For example, if you declare
+
+[source,java]
+----
+@Outgoing("generated-price")
+public Multi<Integer> generate() {
+    ...
+}
+----
+
+and your configuration indicates that the `generated-price` channel uses the `smallrye-kafka` connector, then Quarkus will automatically set the `value.serializer` to Kafka's built-in `IntegerSerializer`.
+
+Similarly, if you declare
+
+[source,java]
+----
+@Incoming("my-kafka-records")
+public void consume(KafkaRecord<Long, byte[]> record) {
+    ...
+}
+----
+
+and your configuration indicates that the `my-kafka-records` channel uses the `smallrye-kafka` connector, then Quarkus will automatically set the `key.deserializer` to Kafka's built-in `LongDeserializer`, as well as the `value.deserializer` to `ByteArrayDeserializer`.
+
+Finally, if you declare
+
+[source,java]
+----
+@Inject
+@Channel("price-create")
+Emitter<Double> priceEmitter;
+----
+
+and your configuration indicates that the `price-create` channel uses the `smallrye-kafka` connector, then Quarkus will automatically set the `value.serializer` to Kafka's built-in `DoubleSerializer`.
+
+The full set of types supported by the serializer/deserializer autodetection is:
+
+* `short` and `java.lang.Short`
+* `int` and `java.lang.Integer`
+* `long` and `java.lang.Long`
+* `float` and `java.lang.Float`
+* `double` and `java.lang.Double`
+* `byte[]`
+* `java.lang.String`
+* `java.util.UUID`
+* `java.nio.ByteBuffer`
+* `org.apache.kafka.common.utils.Bytes`
+* `io.vertx.core.buffer.Buffer`
+* `io.vertx.core.json.JsonObject`
+* `io.vertx.core.json.JsonArray`
+* classes generated from Avro schemata, if Confluent or Apicurio _serde_ is present
```

Comment:
Schema?

Reply:
I wanted to use plural, and both "schemas" and "schemata" are apparently OK, but I can go back to singular.

Reply:
Oh I didn't know the word! I would go with 'schemas'. I don't believe lots of people will understand 'schematas'.

Reply:
`schematas` is far too greek to be generally understandable 😎

Reply:
It's like indexes vs. indices :-) Alright, I'll change it to "schemas".